engine/src/main/java/com/sandboni/core/engine/sta/connector/HttpTemplateConnector.java

```diff
@@ -1,29 +1,62 @@
 import com.sandboni.core.engine.sta.graph.LinkFactory;
 import com.sandboni.core.engine.sta.graph.LinkType;
+import com.sandboni.core.engine.sta.graph.vertex.CucumberVertex;
+import com.sandboni.core.engine.sta.graph.vertex.TestVertex;
+import com.sandboni.core.engine.sta.graph.vertex.Vertex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import java.util.Optional;
+import java.io.*;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.Objects;
 import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
+import static com.sandboni.core.engine.sta.graph.vertex.VertexInitTypes.START_VERTEX;
+
 public class HttpTemplateConnector implements Connector {
 
     private static final Pattern HTTP_PARAMETERS_PATTEN = Pattern.compile("\\{.+?}");
 
+    private static final String CUCUMBER_ENTRY = "CUCUMBER";
+
+    private static final Logger log = LoggerFactory.getLogger(HttpTemplateConnector.class);
+
     @Override
     public void connect(Context context) {
         Set<Link> httpRequests = context.getLinks().filter(l -> l.getLinkType() == LinkType.HTTP_REQUEST).collect(Collectors.toSet());
+
         Set<Link> httpHandlers = context.getLinks().filter(l -> l.getLinkType() == LinkType.HTTP_HANLDER).collect(Collectors.toSet());
 
-        httpRequests.forEach(r -> {
-            Optional<Link> matchingHandler = httpHandlers.stream()
-                    .filter(l ->  isMatch(r.getCallee().getAction(), l.getCaller().getAction())).findAny();
-            matchingHandler.ifPresent(link -> context.addLink(LinkFactory.createInstance(context.getApplicationId(), r.getCallee(), link.getCaller(), LinkType.HTTP_MAP)));
+        httpRequests.forEach(r ->
+                httpHandlers.stream()
+                        .filter(l -> isMatch(r.getCallee().getAction(), l.getCaller().getAction()))
+                        .forEach(link -> context.addLink(LinkFactory.createInstance(context.getApplicationId(), r.getCallee(), link.getCaller(), LinkType.HTTP_MAP)))
+        );
+
+        //handling Seloni json entries
+        JsonEntry[] seloniEntries = getEntriesFromSeloniFile();
+
+        Arrays.stream(seloniEntries).forEach(entry -> {
+            Vertex vertex = createVertex(entry);
+            context.addLink(LinkFactory.createInstance(context.getApplicationId(), START_VERTEX, vertex, LinkType.ENTRY_POINT));
+            entry.getUrls().forEach(url ->
+                httpHandlers.stream()
+                        .filter(l -> isMatch(url, l.getCaller().getAction())).forEach(link -> {
+                    context.addLink(LinkFactory.createInstance(context.getApplicationId(), vertex, link.getCaller(), LinkType.HTTP_MAP_SELONI));
+                    log.debug("[external link] `{}`: {} -> {}  :: {} -> {}", url, link.getCaller().getActor(), link.getCaller().getAction(), link.getCallee().getActor(), link.getCallee().getAction());
```

Comment:
can we follow the same format as other debug statements? I don't think that we need the brakets []

Reply:
ok

Reply:
fixed